src/ecash/service/mod.rs

```diff
@@ -0,0 +1,216 @@
+use crate::ecash;
+
+use std::collections::HashSet;
+use std::convert::TryInto;
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::net::{Ipv4Addr, SocketAddr};
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use futures::StreamExt;
+use tracing::{debug, error, warn};
+
+mod handlers;
+
+#[derive(Clone)]
+struct MintState {
+    ln: ln_rs::Ln,
+    mint: Arc<tokio::sync::Mutex<cdk::mint::Mint>>,
+}
+
+pub async fn mint_service(
+    config_file_arg: String,
+) -> anyhow::Result<tokio::task::JoinHandle<()>> {
+    let ecash_config = match ecash::config::EcashConfig::new(&Some(config_file_arg)) {
+        Ok(mint_settings) => mint_settings,
+        Err(e) => {
+            error!("{}. Halting.", e);
+            std::process::exit(1);
+        }
+    };
+
+    // Create db_path parent directory if it doesn't exist
+    guarantee_parent_path(&ecash_config.mint.db_path)?;
+    let localstore = cdk::mint::RedbLocalStore::new(
+        ecash_config
+            .mint
+            .db_path
+            .to_str()
+            .expect("PathBuf always converts to &str"),
+    )?;
+
+    let mint = cdk::mint::Mint::new(
+        Arc::new(localstore),                                       // localstore
+        bip39::Mnemonic::from_str(&ecash_config.mint.mnemonic)?,    // mnemonic
+        HashSet::new(),                                             // keysets_info
+        cdk::amount::Amount::ZERO,                                  // min_fee_reserve
+        0.0,                                                        // percent_fee_reserve
+    )
+    .await?;
+
+    debug!("Mint created");
+
+    let cln_socket = expand_path(
+        ecash_config
+            .ln
+            .cln_path
+            .clone()
+            .ok_or(anyhow::anyhow!("cln socket not defined"))?
+            .to_str()
+            .ok_or(anyhow::anyhow!("cln socket not defined"))?,
+    )
+    .ok_or(anyhow::anyhow!("cln socket not defined"))?;
+
+    let last_pay_path = PathBuf::from_str(&ecash_config.mint.last_pay_path.clone())?;
+
+    // Create last_pay_path if it doesn't exist
+    match fs::metadata(&last_pay_path) {
+        Ok(_) => (),
+        Err(_e) => {
+            // Create the parent directory if it doesn't exist
+            fs::create_dir_all(last_pay_path.parent().unwrap())?;
```

Comment:
```suggestion
            fs::create_dir_all(last_pay_path.parent()?)?;
```

Reply:
simply replacing with `?` breaks compilation.

c5e916fad5929702fa1b7827c123d63169bef5a7 did the trick

Reply:
I'm sorry, you also need to use .context, something like:
```rust
let parent_path = last_pay_path
                .parent()
                .context("Parent directory not found")?;
fs::create_dir_all(parent_path)?;
```

Or to give a better context to the caller:
```rust
let parent_path = last_pay_path
                .parent()
                .with_context(|| format!("Parent directory not found for {last_pay_path:?}"))?;
fs::create_dir_all(parent_path)?;
```